src/main/java/io/github/syst3ms/skriptparser/expressions/ExecExprListOperators.java

```diff
@@ -32,44 +32,74 @@
  * @type EXPRESSION
  * @pattern pop %objects%
  * @pattern (shift|poll) %objects%
- * @pattern splice %objects% from %integer% [1:to %integer%]
+ * @pattern splice %objects% (from %integer% to %integer%|starting (at|from) %integer%|up to %integer%) [[with] step %integer%]
  * @since ALPHA
  * @author Mwexim
  * @see ExprElement
  */
 public class ExecExprListOperators extends ExecutableExpression<Object> {
-
 	static {
 		Parser.getMainRegistration().addSelfRegisteringElement(
 				ExecExprListOperators.class,
 				Object.class,
 				false,
+				"extract [the] (0:last|1:first|2:%integer%(st|nd|rd|th)) element out [of] %objects%",
 				"pop %objects%",
 				"(shift|poll) %objects%",
-				"splice %objects% from %integer% [1:to %integer%]"
+				"splice %objects% (0:from %integer% to %integer%|1:starting (at|from) %integer%|2:up to %integer%) [[with] step %integer%]"
```

Comment:
Should the `splice` keyword be mandatory? Sure, for an effect it wouldn't make sense, but for an expression I wonder whether splicing is common enough for it to be required. I'm not sure.

Having "until" as an alternative to "up to" would also be desirable, because (1) it's a common English construction, and (2) it's less unwieldy when using negative steps.

Reply:
I think removing the `splice` keyword might prove to be more unwieldy than before. Without the keyword, no one really knows what this method does. People may think that it just returns the values in that range, but they also get removed from the list, which may not be obvious at first.

Reply:
There should be a way to splice a list without actually changing it. Perhaps with two separate classes? I don't really like non-obvious side-effects in expressions.

Reply:
It'd be non-obvious if we'd remove the `splice` keyword. Now people know that this happens. We can add these expressions later in a separate class or add them to ExprMutableList for example, but that's currently something that I will not add in this particular PR.